File: Basic Java/TestingSystem_Assignment/src/com/vti/backend/TestingSystem_Assignment_4/CanBo.java
package com.vti.backend.TestingSystem_Assignment_4;

public class CanBo {
    private long id;
    private String fullName;
    private int age;
    private Gender gender;
    private String address;

    public CanBo() {}

    public CanBo(long id, String fullName, int age, Gender gender, String address) {
        this.id = id;
        this.fullName = fullName;
        this.age = age;
        this.gender = gender;
        this.address = address;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public Gender getGender() {
        return gender;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }
@Override
    public String toString() {
        return "\nCán bộ:" + "\nTên: " + fullName + "\nTuổi: " + age + "\nGiới tính: " + gender + "\nĐịa chỉ: " + address;
    }
}

File: ReactJS/Assignment_02/Lab_02/CanBo.ts

export class CanBo {

    private static counter: number = 0;

    private id: number;
    private fullName: string;
    private age: number;
    private gender: Gender;
    private address: string;

    constructor(fullName: string, age: number, gender: Gender, address: string) {
        this.id = ++CanBo.counter;
        this.fullName = fullName;
        this.age = age;
        this.gender = gender;
        this.address = address;
    }


    /**
     * Getter id
     * @return {number}
     */
	public getId(): number {
		return this.id;
	}

    /**
     * Getter fullName
     * @return {string}
     */
	public getFullName(): string {
		return this.fullName;
	}

    /**
     * Getter age
     * @return {number}
     */
	public getAge(): number {
		return this.age;
	}

    /**
     * Getter gender
     * @return {Gender}
     */
	public getGender(): Gender {
		return this.gender;
	}

    /**
     * Getter address
     * @return {string}
     */
	public getAddress(): string {
		return this.address;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public setId(value: number) {
		this.id = value;
	}

    /**
     * Setter fullName
     * @param {string} value
     */
	public setFullName(value: string) {
		this.fullName = value;
	}

    /**
     * Setter age
     * @param {number} value
     */
	public setAge(value: number) {
		this.age = value;
	}

    /**
     * Setter gender
     * @param {Gender} value
     */
	public setGender(value: Gender) {
		this.gender = value;
	}

    /**
     * Setter address
     * @param {string} value
     */
	public setAddress(value: string) {
		this.address = value;
	}

    public displayDetails(){
        console.log( `
            ID: ${this.getId()},
            FullName: ${this.getFullName()}, 
            Age: ${this.getAge()}, 
            Gender: ${this.getGender()},
            Address: ${this.getAddress()}
        `);
    }
}



export enum Gender {
    MALE, FEMALE, OTHER
}

